File: src/test/java/org/zenbot/szolnok/timetable/batch/step/readurls/StringResourcesInMemoryStorageTest.java
package org.zenbot.szolnok.timetable.batch.step.readurls;

import org.junit.Before;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class StringResourcesInMemoryStorageTest {

    private static final String URL_1 = "URL_1";
    private static final String URL_2 = "URL_2";
    private StringResourcesInMemoryStorage testSubject;

    @Before
    public void setUp() {
        testSubject = new StringResourcesInMemoryStorage();
    }

    @Test
    public void addNullUrlShouldNotBeAdded() {
        // GIVEN
        testSubject.addUrl(null);

        // WHEN
        String result = testSubject.get();

        // THEN
        assertThat(result)
                .isNull();
    }

    @Test
    public void getShouldReturnWithTheFirstThenRemoveThat() {
        // GIVEN
        testSubject.addUrl(URL_1);
        testSubject.addUrl(URL_2);

        // WHEN
        String first = testSubject.get();
        String second = testSubject.get();
        String third = testSubject.get();

        // THEN
        assertThat(first).isNotNull().isEqualTo(URL_1);
        assertThat(second).isNotNull().isEqualTo(URL_2);
        assertThat(third).isNull();
    }

}

File: szolnok-timetable-backend-batch-utils/src/test/kotlin/org/zenbot/szolnok/timetable/backend/batch/utils/common/service/StringResourcesInMemoryStorageTest.kt
package org.zenbot.szolnok.timetable.backend.batch.utils.common.service

import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test

class StringResourcesInMemoryStorageTest {

    private lateinit var testSubject: StringResourcesInMemoryStorage

    @Before
    fun setUp() {
        testSubject = StringResourcesInMemoryStorage()
    }

    @Test
    fun `addUrl should add to the list and get should rmeove the first`() {
        // GIVEN
        val url1 = "URL1"
        val url2 = "URL2"
        testSubject.addUrl(null)
        testSubject.addUrl(url1)
        testSubject.addUrl(url2)

        // WHEN
        val get1 = testSubject.get()
        val get2 = testSubject.get()
        val get3 = testSubject.get()

        // THEN
        assertThat(get1).isNotNull().isSameAs(url1)
        assertThat(get2).isNotNull().isSameAs(url2)
        assertThat(get3).isNull()
    }
}